src/logistration/Logistration.jsx

```diff
@@ -116,7 +117,7 @@ const Logistration = (props) => {
                     <Tab title={tabTitle} eventKey={selectedPage === LOGIN_PAGE ? LOGIN_PAGE : REGISTER_PAGE} />
                   </Tabs>
                 )
-                : (!isValidTpaHint() && (
+                : (!isValidTpaHint() && !hideRegistrationLink && (
                   <Tabs defaultActiveKey={selectedPage} id="controlled-tab" onSelect={handleOnSelect}>
                     <Tab title={formatMessage(messages['logistration.register'])} eventKey={REGISTER_PAGE} />
                     <Tab title={formatMessage(messages['logistration.sign.in'])} eventKey={LOGIN_PAGE} />
```

Comment:
```suggestion
                : (!isValidTpaHint() && (
                  <Tabs defaultActiveKey={selectedPage} id="controlled-tab" onSelect={handleOnSelect}>
                    <Tab title={formatMessage(messages['logistration.sign.in'])} eventKey={LOGIN_PAGE} />
                    {!hideRegistrationLink || selectedPage === REGISTER_PAGE && (<Tab title={formatMessage(messages['logistration.register'])} eventKey={REGISTER_PAGE} />)}
```

Instead of adding a new h3 element, can we hide/show register tab based on the flag and url, something like above. This way users have a way to go to sign-in page from register page like they would normally do if both tabs are enabled.

Reply:
@navinkarkera  Thanks for the review.  

Well spotted! You put your finger on the dilemma I was facing before opting for the new h3 element. My decision was motivated by the following: 

1. In the majority of cases, only the `login` page will be accessible, and I tried to replicate the design of the scenario `ALLOW_PUBLIC_ACCOUNT_CREATION == false`. I find this design cleaner because if we leave a single tab, it remains clickable but also invites the question why is it a tab if there are no other tabs?
2. I considered the problem of users not having a way to go to `login` from `register` and I convinced myself that `register` in the scenario of `SHOW_REGISTRATION_LINKS == false` if more of a [fallback](https://github.com/openedx/edx-platform/pull/32783#issuecomment-1701037624) than the main path. In addition and this is purely subjective, the disappearance of the `register` tab (if we opt for keeping the tabs) when we make the transition `register => login` seemed off. 

However, your suggestion makes the code cleaner and is easier to reason about. I am happy to go either way.

Reply:
> I tried to replicate the design of the scenario ALLOW_PUBLIC_ACCOUNT_CREATION == false

@CefBoud Ahh, makes sense. I did not notice that before. Thanks for the detailed explanantion.

I agree with your approach so propose to keep the `h3` element and leave this thread open to get suggestions from upstream reviewers.

Reply:
@navinkarkera Ok, great. Thank you for taking the time.
@mubbsharanwar, I am taking the liberty to ping you as an FYI as you were the one who [requested the changes](https://github.com/openedx/edx-platform/pull/32783#issuecomment-1702579527).